packages/ragbits-document-search/src/ragbits/document_search/_main.py

```diff
@@ -89,7 +90,15 @@ async def ingest_document(
             document_processor: The document processor to use. If not provided, the document processor will be
                 determined based on the document metadata.
         """
-        document_meta = document if isinstance(document, DocumentMeta) else document.metadata
+
+        if isinstance(document, Source):
+            local_path = await document.fetch()
+            document_meta = DocumentMeta.from_local_path(local_path)
```

Comment:
hmm, it's not going to work like that. 

When ingesting document from a given source (for example `gcs`) we should have that information persisted along with DocumentMeta. `DocumentMeta.from_local_path` creates new `LocalFileSource` so it loses that important piece of info

Reply:
ok, good point, changed